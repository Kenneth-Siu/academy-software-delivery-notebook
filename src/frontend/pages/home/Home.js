import React from "react";
import { Link } from "react-router-dom";
import "./Home.scss";

export default function Home() {
    return (
        <main className="home-page">
            <h1>Notebook Cookbook</h1>
            <div className="links">
                <Link to="/shopping-list">
                    <span className="emoji">üõí</span>
                    <span className="text">Shopping List</span>
                </Link>
                <Link to="/cookbook">
                    <span className="emoji">üç≥</span>
                    <span className="text">Cookbook</span>
                </Link>
            </div>
        </main>
    );
}
