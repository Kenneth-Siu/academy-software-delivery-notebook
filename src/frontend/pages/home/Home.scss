@import "../../components/colors.scss";

.home-page {
    padding-top: 1rem;

    h1 {
        text-align: center;
        font-size: 2rem;
        margin-bottom: 1rem;
    }

    .links {
        display: flex;
        flex-wrap: wrap;
        justify-content: center;

        a {
            text-decoration: none;
            color: $color-1;
            width: 20rem;
            height: 20rem;
            flex-grow: 0;
            flex-shrink: 0;
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            border: solid $color-1 1px;
            border-radius: 2rem;
            margin: 0.8rem;

            .emoji {
                font-size: 8rem;
                margin-bottom: 1.5rem;
            }

            .text {
                font-size: 2rem;
            }
        }
    }
}
