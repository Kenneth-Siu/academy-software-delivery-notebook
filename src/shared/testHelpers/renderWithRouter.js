import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

export default function renderWithRouter(component) {
    return render(<MemoryRouter>{component}</MemoryRouter>);
}
