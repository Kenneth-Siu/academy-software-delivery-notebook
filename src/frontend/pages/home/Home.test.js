import React from "react";
import Home from "./Home";
import renderWithRouter from "../../../shared/testHelpers/renderWithRouter";

describe("Home", () => {
    test("renders", () => {
        const tree = renderWithRouter(<Home />);
        expect(tree.container).toMatchSnapshot();
    });
});
