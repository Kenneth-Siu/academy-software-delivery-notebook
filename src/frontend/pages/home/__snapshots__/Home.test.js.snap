// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<div>
  <main
    class="home-page"
  >
    <h1>
      Notebook Cookbook
    </h1>
    <div
      class="links"
    >
      <a
        href="/shopping-list"
      >
        <span
          class="emoji"
        >
          üõí
        </span>
        <span
          class="text"
        >
          Shopping List
        </span>
      </a>
      <a
        href="/cookbook"
      >
        <span
          class="emoji"
        >
          üç≥
        </span>
        <span
          class="text"
        >
          Cookbook
        </span>
      </a>
    </div>
  </main>
</div>
`;
